synapse/util/versionstring.py

```diff
@@ -44,71 +46,36 @@ def get_version_string(module: ModuleType) -> str:
     version_string = module.__version__  # type: ignore[attr-defined]
 
     try:
-        null = open(os.devnull, "w")
-        cwd = os.path.dirname(os.path.abspath(module.__file__))
+        with open(os.devnull, "w") as null:
+            cwd = os.path.dirname(os.path.abspath(module.__file__))
+
+            def _run_git_command(prefix: str, *params: str) -> str:
+                try:
+                    result = (
+                        subprocess.check_output(["git", *params], stderr=null, cwd=cwd)
```

Comment:
It might be more idiomatic to use [`stderr=subprocess.DEVNULL`](https://docs.python.org/3/library/subprocess.html?highlight=subprocess#subprocess.DEVNULL)? I'd have to convince myself that it doesn't open and leak a file descriptor though. (I'd expect it doesn't, but...)

Reply:
That might be better, I'll take a look! Thanks!